src/Server.h

```diff
@@ -0,0 +1,19 @@
+#pragma once
+#include "common/ServerSocket.h"
+#include "common/Socket.h"
+
+class Server
+{
+public:
+    explicit Server(std::unique_ptr<ServerSocket> server_socket, size_t max_connections);
```

Comment:
would rather `server_socket` would be encapsulated inside this class (self),
the API shouldn't expose this